canonical-web-and-design/maas-ui
# feat(ui): update tag form

## Done

- Add a form to update a tag.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the tag list and click one of the edit buttons. You should see the edit tag form.
- Click cancel and you should go back to the tag list.
- Go to the details page for a tag and click the edit button, you should see the form.
- Click cancel and you should go back to the details page.
- Go to the edit form and copy the URL and open it in a new tab.
- Click cancel and you should go to the tag details.
- Go to the edit form for a tag that is applied to a deployed machine and change the kernel options. You should see a warning appear about the deployed machines.
- Go to the form for a tag without a definition (i.e. manual) and you should have a message instead of the definition field.
- Go to the form for a tag with a definition and change the definition. You should get a warning about the tag being unassigned from non-matching machines.
- Remove the definition completely and you should get an error about not being able to remove the definition.
- Make some changes to a tag and click save and the changes should get persisted.


## Fixes

Fixes: canonical-web-and-design/app-tribe#706.

## Screenshot

<img width="1460" alt="Screen Shot 2022-03-25 at 11 52 31 am" src="https://user-images.githubusercontent.com/361637/160043057-8dde9e96-f6ad-4c9f-9a11-22cd4bac4f82.png">
